localstack/services/lambda_/invocation/event_manager.py

```diff
@@ -7,34 +7,277 @@
+        self.queues.remove(queue)
+
+    def do_update_all_queues(self):
+        for queue in self.queues:
+            try:
+                queue.requeue_inflight_messages()
+            except Exception:
+                LOG.exception("error re-queueing inflight messages")
+
+            try:
+                queue.enqueue_delayed_messages()
+            except Exception:
+                LOG.exception("error enqueueing delayed messages")
+
+            if config.SQS_ENABLE_MESSAGE_RETENTION_PERIOD:
+                try:
+                    queue.remove_expired_messages()
+                except Exception:
+                    LOG.exception("error removing expired messages")
+
+    def start(self):
+        with self.mutex:
+            if self.thread:
+                return
+
+            self.scheduler = Scheduler()
+            self.scheduler.schedule(self.do_update_all_queues, period=1)
+
+            def _run(*_args):
+                self.scheduler.run()
+
+            self.thread = start_thread(_run, name="sqs-queue-update-worker")
+
+    def stop(self):
+        with self.mutex:
+            if self.scheduler:
+                self.scheduler.close()
+
+            if self.thread:
+                self.thread.stop()
+
+            self.thread = None
+            self.scheduler = None
+
+
+class QueueManager:
+    queues: dict[str, StandardQueue]
+    queue_lock: threading.RLock
+    queue_update_worker: QueueUpdateWorker
+
+    def __init__(self):
+        self.queues = {}
+        self.queue_lock = threading.RLock()
+        self.queue_update_worker = QueueUpdateWorker()
+        self.queue_update_worker.start()
+
+    def get_queue(self, queue_name: str):
+        if queue_name not in self.queues:
+            raise ValueError("Queue not available")
+        return self.queues[queue_name]
+
+    def create_queue(self, queue_name: str) -> SqsQueue:
+        """
+        Creates a queue.
+        :param queue_name: Queue name, has to be unique
+        :return: Queue Object
+        """
+        with self.queue_lock:
+            if queue_name in self.queues:
+                return self.queues[queue_name]
+
+            # Truncate function name to ensure queue name limit of max 80 characters
+            queue = StandardQueue(
+                name=queue_name,
+                region="us-east-1",
+                account_id=config.INTERNAL_RESOURCE_ACCOUNT,
+            )
+            self.queues[queue_name] = queue
+            self.queue_update_worker.add_queue(queue)
+        return queue
+
+    def delete_queue(self, queue_name: str) -> None:
+        with self.queue_lock:
+            if queue_name not in self.queues:
+                raise ValueError("Queue not available")
+
+            queue = self.queues.pop(queue_name)
+            self.queue_update_worker.remove_queue(queue)
+
+
+if config.LAMBDA_EVENTS_INTERNAL_SQS:
+    QUEUE_MANAGER = QueueManager()
+else:
+    QUEUE_MANAGER = None
+
+
+class FakeSqsClient:
```

Comment:
can we remove this and instead inject the queue manager as dependency into the fake client?